soh/include/macros.h

```diff
@@ -204,6 +204,14 @@ extern GraphicsContext* __gfxCtx;
 #define ALIGNED8
 #endif
 
-#define SEG_ADDR(seg, addr) (addr | (seg << 24) | 0xF0000000)
+#define SEG_ADDR(seg, addr) (addr | (seg << 24) | 1)
```

Comment:
Why is `0xF0000000` flipping to `0x00000001`, is there an endian-ness thing happening?

Reply:
This is a design issue in the code to differentiate a segmented and non-segmented address. This original hack was based on the assumption that the higher bits cannot be valid addresses. However this is only true on Windows. On linux this may completely break (and already has in the past) if you're not lucky enough with the heap layout.
The fix here is to used the lower bit instead which is never used for segmented addresses because they're all alligned.

Reply:
Thanks for the responses. This is helping me a lot in terms of understanding this codebase.